ansible/roles/csm.ncn.validate_plan/tasks/main.yml

```diff
@@ -0,0 +1,38 @@
+# Copyright 2021 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# (MIT License)
+
+---
+- include_role:
+    name: csm.ncn.download_utils
+- include_role:
+    name: csm.ncn.standardize_hostnames
+
+- name: Verify cluster operation assuming inventory of hosts goes away
+  delegate_to: localhost
+  run_once: true
+  environment:
+    SLS_BASE_URL: "{{ sls_url }}"
+    BSS_BASE_URL: "{{ bss_url }}"
+    HSM_BASE_URL: "{{ hsm_url }}"
+    TOKEN: "{{ token }}"
+  command: csi automate ncn preflight --action verify-loss-acceptable
```

Comment:
I don't understand what this command is doing. It sounds like we're trying to see if the cluster survives if we killed all of the target hosts. Maybe I'm misreading, but I thought all_target_hostsnames included all of the hosts in the cluster. Is this interpretation incorrect? Because I can't see the cluster surviving, if you killed everything.

Reply:
I can see how that would be confusing...basically the intention with this rebuild logic is to never (well, never say never just not supported yet) allow *all* of the NCNs to be targeted at the same time. More concretely a rebuild invocation is expected to be something like this:

```
cray cfs sessions create --name ncn-rebuild-ncnw00567 --configuration-name rebuild-ncn --ansible-verbosity 1 --ansible-limit x3001c0s31b0n0,x3000c0s38b0n0,x3000c0s36b0n0
```

So in a sense the intention isn't for this logic to ever be run against the entire list of Management nodes and that's pretty much the intention of this check.